server/SocketManager/socketmanager.js

```diff
@@ -1,11 +1,14 @@
 let sockets = {};
+let socket_user_map = {};
 
-const updateSocket = (usertoken, clientsocket) => {
+const updateSocket = (usertoken, clientsocket, user_id) => {
   sockets[usertoken] = clientsocket;
+  socket_user_map[usertoken] = user_id;
```

Comment:
The `updateSocket` function now takes an additional parameter `user_id`, which is a good addition for tracking users. However, ensure that `user_id` is validated before being stored in `socket_user_map` to prevent potential issues with invalid or undefined values. Consider adding a check for this.
```diff
+  if (!user_id) {
+    throw new Error('Invalid user_id');
+  }
```
*This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*